packages/@aws-cdk/aws-stepfunctions-tasks/lib/ecs/run-task.ts

```diff
@@ -0,0 +1,376 @@
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as ecs from '@aws-cdk/aws-ecs';
+import * as iam from '@aws-cdk/aws-iam';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import * as cdk from '@aws-cdk/core';
+import { ContainerOverride } from '..';
+import { integrationResourceArn, validatePatternSupported } from '../private/task-utils';
+
+/**
+ * Basic properties for ECS Tasks
+ */
+export interface EcsRunTaskProps extends sfn.TaskStateBaseProps {
+  /**
+   * The ECS cluster to run the task on
+   */
+  readonly cluster: ecs.ICluster;
+
+  /**
+   * [disable-awslint:ref-via-interface]
+   * Task Definition used for running tasks in the service.
+   *
+   * Note: this must be TaskDefinition, and not ITaskDefinition,
+   * as it requires properties that are not known for imported task definitions
+   */
+  readonly taskDefinition: ecs.TaskDefinition;
```

Comment:
The comment explains why it shouldn't be `ITaskDefinition` - does it help to spell out the specific properties that we require that are not captured in `ITaskDefinition`?

Reply:
I thought, using `ITaskDefinition` would make it nice so customers can use imported task definitions.

Fine to leave it as is if it's too hard to switch.

Reply:
It would definitely be nicer and to avoid having the linter exception as well. I've tried to do the same for all the other service integrations (use `IBlah`). 

In this case, I couldn't think of a nicer way forward without some changes to ITaskDefinition itself or requesting additional params